extensions/sql-migration/src/models/stateMachine.ts

```diff
@@ -177,27 +184,160 @@ export class MigrationStateModel implements Model, vscode.Disposable {
 
 	public async getDatabaseAssessments(): Promise<ServerAssessement> {
 		const ownerUri = await azdata.connection.getUriForConnection(this.sourceConnectionId);
-		// stress test backend & dialog component
-		const assessmentResults = await this.migrationService.getAssessments(
-			ownerUri,
-			this._databaseAssessment
-		);
-		const dbAssessments = assessmentResults?.assessmentResult.databases.filter(d => !this.excludeDbs.includes(d.name)).map(d => {
-			return {
-				name: d.name,
-				issues: d.items.filter(i => i.appliesToMigrationTargetPlatform === MigrationTargetType.SQLMI) ?? []
-			};
-		});
+		this._assessmentApiResponse = (await this.migrationService.getAssessments(ownerUri, this._databaseAssessment))!;
 		this._assessmentResults = {
-			issues: assessmentResults?.assessmentResult.items?.filter(i => i.appliesToMigrationTargetPlatform === MigrationTargetType.SQLMI) ?? [],
-			databaseAssessments: dbAssessments! ?? []
+			issues: this._assessmentApiResponse.assessmentResult.items,
+			databaseAssessments: this._assessmentApiResponse.assessmentResult.databases.map(d => {
+				return {
+					name: d.name,
+					issues: d.items
+				};
+			})
 		};
-
+		this.generateAssessmentTelemetry();
 		return this._assessmentResults;
 	}
 
-	public findDatabaseAssessments(databaseName: string): mssql.SqlMigrationAssessmentResultItem[] | undefined {
-		return this._assessmentResults.databaseAssessments.find(databaseAsssessment => databaseAsssessment.name === databaseName)?.issues;
+	private async generateAssessmentTelemetry(): Promise<void> {
+		try {
+			this._assessmentResults.issues.forEach(i => {
```

Comment:
How many issues are you expecting users to be getting? We want to be careful that we're not sending too many events all at once if we can avoid it.

Reply:
Issues can range from a few (1-2) to many at a single time (100+)

Reply:
SInce we are just sending a map it should be 10-12 issues at the most